src/ServiceInsight.Tests/XamlIconsTests.cs

```diff
@@ -0,0 +1,170 @@
+            new FrameworkElement();
+            Application.ResourceAssembly = typeof(App).Assembly;
+        }
+
+        private static ResourceDictionary GetXamlIconsResourceDictionary()
+            => new ResourceDictionary { Source = new Uri("/ServiceInsight;component/Images/Xaml/XamlIcons.xaml", UriKind.Relative) };
+
+        private static ContentControl CreateDisplayFrame(string resourceName, UIElement displayElement)
+        {
+            var control = new ContentControl();
+            var dock = new DockPanel();
+            var text = new TextBlock();
+
+            control.Width = 500;
+            control.Height = 500;
+            control.Content = dock;
+            DockPanel.SetDock(text, Dock.Top);
+            dock.Children.Add(text);
+            dock.Children.Add(displayElement);
+            text.FontSize = 24;
+            text.FontWeight = FontWeights.Bold;
+            text.HorizontalAlignment = HorizontalAlignment.Center;
+
+            text.Text = resourceName;
+
+            var xamlIcons = GetXamlIconsResourceDictionary();
+            control.Resources.MergedDictionaries.Add(xamlIcons);
+
+            return control;
+        }
+
+        private static IEnumerable GetTestCases<TValue>()
+        {
+            IEnumerable<TestCaseData> results = null;
+            var wait = new SemaphoreSlim(0);
+            var ts = new ThreadStart(() =>
+            {
+                InitWPFResourcesSystem();
+
+                var xamlIcons = GetXamlIconsResourceDictionary();
+
+                results = xamlIcons.Cast<DictionaryEntry>()
+                    .Where(e => e.Value is TValue)
+                    .Select(e => new TestCaseData((string)e.Key).SetName((string)e.Key))
+                    .ToArray();
+
+                wait.Release();
+            });
+            var t = new Thread(ts);
+            t.SetApartmentState(ApartmentState.STA);
+            t.Start();
+            wait.Wait();
+            return results;
+        }
+
+        private static void PrepareForRender(UIElement control)
+        {
+            // From http://stackoverflow.com/a/2596035
+            control.Measure(new Size(double.PositiveInfinity, double.PositiveInfinity));
+            control.Arrange(new Rect(control.DesiredSize));
+
+            control.Dispatcher.Invoke(() => { }, DispatcherPriority.Loaded);
+        }
+
+        public static IEnumerable GetControlTemplates => GetTestCases<ControlTemplate>();
+
+        public static IEnumerable GetDrawingImages => GetTestCases<DrawingImage>();
+
+        public static IEnumerable GetGeometries => GetTestCases<Geometry>();
+
+        [Test]
+        [Explicit]
+        [TestCaseSource(typeof(XamlIconsTests), nameof(GetControlTemplates))]
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public void ControlTemplateTest(string resourceName)
+        {
+            var icon = new ContentControl { Foreground = Brushes.Green };
```

Comment:
It's not a color we use in the coloured icons. It's just for contrast. Any colour other than white can be used here.